src/ngMock/angular-mocks.js

```diff
@@ -1187,6 +1188,10 @@ function createHttpBackendMock($rootScope, $timeout, $delegate, $browser) {
       }
     }
 
+    if (notCalledExpectations.some(function(exp) { return exp.match(method, url, data, headers); })) {
+      throw new Error('Request should not be called: ' + method + ' ' + url);
```

Comment:
I'm not sure whether this is necessary, but you could include data and headers in the error description.